### ISSUE TITLE
C# unable to convert dictionary to JSON

### ISSUE THREAD

**Issue by TwistedTwigleg** [godotengine/godot#23811]
**Godot version:**
Godot 3.0.6 Mono

**OS/device including version:**
Ubuntu Linux 16.04 LTS

**Issue description:**

C# in Godot cannot seem to convert dictionary objects to JSON using `JSON.print`.

A user (Scozor) on the [Godot Developer forums](https://godotdevelopers.org/forum/discussion/20235/dictionary-and-json-error) made a post saying that they were unable to save a dictionary to a file following the [Saving Games](https://docs.godotengine.org/en/3.0/tutorials/io/saving_games.html) tutorial on the documentation.

Me and Scozor have both tried to find a way to save a dictionary following the instructions in the documentation with no results and we get almost the exact same error (slightly different due to using different operating systems).

It appears there is something wrong in converting the dictionary to a JSON string, as the project crashes when trying to convert dictionaries. Converting a string to JSON works fine, as does storing a string to a file.

I get the following error: `drivers/unix/stream_peer_tcp_posix.cpp:276 - Server disconnected!` when converting a Dictionary into a JSON string. The user on the forums gets the following error: `drivers/windows/stream_peer_tcp_winstock.cpp:203 - Server disconnected!`, which seems like the same issue, just for different operating systems

**Steps to reproduce:**
Try to save a dictionary to a file using `JSON.Print`.

**Minimal reproduction project:**
[CSharp_Save_and_Load_Test.zip](https://github.com/godotengine/godot/files/2593174/CSharp_Save_and_Load_Test.zip)

**Comment by TwistedTwigleg** [comment]
> In 3.0.6 you should be using Godot.Dictionary and Godot.Array (note these were moved to the Godot.Collections namespace in later versions). Does the bug persist when using those instead?

Thanks for the fast reply.

I changed `Dictionary<string, object>` to `Godot.Dictionary<string, object>` and the project still compiles, but unfortunately the error is still the same and it still crashes without writing to the file.

**Comment by andfaxle** [comment]
I´ve got the same error:

```
` drivers/windows/stream_peer_tcp_winsock.cpp:203 - Server disconnected!
```
Here is my Code:
```
Godot.Dictionary<object, object> dic = new Godot.Dictionary<object, object>();
dic.Add("hallo","Dies ist ein test");
string test = JSON.Print(dic );
```
Godot version:
Godot 3.0.6 Mono

OS:
Windows 10

Thanks for your help : )

**Comment by PJB3005** [comment]
Pretty sure this bug is fixed by #20706, but it hasn't been cherry picked onto stable.

**Comment by akien-mga** [comment]
Assuming fixed by #20706 then. No updates are planned for Mono support in the 3.0 branch.
